server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorListenerImpl.java

```diff
@@ -399,14 +404,40 @@ private Locality retrievePosition(@NonNull ParserRuleContext ctx) {
 
   @NonNull
   private Function<Token, Locality> toPosition() {
-    return token ->
-        Locality.builder()
-            .uri(documentUri)
-            .copybookId(retrieveCopybookId())
-            .range(new Range(getStartPosition(token), getStopPosition(token)))
-            .token(token.getText())
-            .recognizer(GrammarPreprocessorListenerImpl.class)
-            .build();
+    return token -> {
+      Position startPosition;
+      Position stopPosition;
+      if (isTokenRepositioned(token)) {
+        CobolLine continueCobolLine = positionCorrectionMap.get(token.getLine() - 1);
```

Comment:
What will happen if the token follows a literal that consists of more than one continuation line?

Reply:
Correct. Logic fails here. I will draft this PR.